### PR TITLE
log in utc

### PR DESCRIPTION
**Pull request by seanstory**
I noticed log lines like:
```
[FMWK][14:25:44][DEBUG] [job_scheduling_service] Scheduler woke up at 2024-07-09 19:25:44.085357+00:00.
```

notice that the logging framework timestamp is `14:25:44` (my local time) but the connectors logic understands that the time should be reasoned about in UTC (`19:25:44`). 

This discrepancy makes it hard to get scheduling right. If I just look at the _majority_ of timestamps that I see in the logs, I'd assume that the scheduling logic should use my local time, and not UTC time.

This PR fixes the issue by overwriting `logging.Formatter.format_time()` for our Formatter class.

It also simplifies the logic for changing the style depending on log level so that function overwrites work. 

#### Pre-Review Checklist
- [x] this PR does NOT contain credentials of any kind, such as API keys or username/passwords (double check `config.yml.example`)
- [x] this PR has a meaningful title
- [x] this PR has a thorough description
- [ ] Covered the changes with automated tests
- [x] Tested the changes locally
- [x] Added a label for each target release version (example: `v7.13.2`, `v7.14.0`, `v8.0.0`)

## Release Note

Connector logs now use UTC timestamps, instead of machine-local timestamps. This only impacts logging output.

### CODE REVIEW

File: connectors/logger.py [hunk 1]
```diff
@@ -32,41 +35,38 @@ class ColorFormatter(logging.Formatter):
...
-    def critical_formatter(self):
-        return logging.Formatter(
-            fmt=self.BOLD_RED + self.custom_format + self.RESET, datefmt=self.DATE_FMT
-        )
+        super().__init__(datefmt=self.DATE_FMT)
+
+    def converter(self, timestamp):
+        tzinfo = tzlocal.get_localzone()
+        dt = datetime.datetime.fromtimestamp(timestamp, tzinfo)
+        return dt.astimezone(pytz.UTC)
```
Comment (jedrazb) [inline connectors/logger.py:43]: I think we can avoid adding `pytz` to the mix, `datetime` lib supports all we need, see example:
https://github.com/elastic/connectors/blob/644c4e018043e24e529b8dadce7665e7da47f5a1/connectors/utils.py#L97-L102

File: connectors/logger.py [hunk 2]
```diff
@@ -32,41 +35,38 @@ class ColorFormatter(logging.Formatter):
...
+            try:
+                s = dt.isoformat(timespec="milliseconds")
+            except TypeError:
+                s = dt.isoformat()
+        return s
 
     def format(self, record):  # noqa: A003
-        formatter = getattr(self, f"{record.levelname.lower()}_formatter")
-        return formatter.format(record)
+        # record.asctime = self.formatTime(record, self.DATE_FMT)
```
Comment (jedrazb) [inline connectors/logger.py:58]: Should this line be removed?

File: connectors/logger.py [hunk 3]
```diff
@@ -32,41 +35,38 @@ class ColorFormatter(logging.Formatter):
...
+        if record.levelno == logging.DEBUG:
```
Comment (jedrazb) [inline connectors/logger.py:59]: Nice! Less is more! ✨

File: connectors/logger.py (hunk 2 above)
Comment (seanstory) [reply connectors/logger.py]: thanks! done.

**Review by jedrazb — APPROVED** [review]

**Pull request merged**